### ISSUE TITLE
Not able to compile any longer

### ISSUE THREAD

**Issue by veganvelociraptor** [Frogging-Family/vulkan-icd-loader-git#4]
I managed  to compile it about a month ago, but not any longer:

```
==> Making package: vulkan-icd-loader-git 1.3.224.r0.ga0ab2f162-1 (Thu Aug 11 10:54:46 2022)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
-> Updating Vulkan-Loader git repo...
==> Validating source files with sha256sums...
Vulkan-Loader ... Skipped
==> Extracting sources...
-> Creating working copy of Vulkan-Loader git repo...
Cloning into 'Vulkan-Loader'...
done.
==> Starting pkgver()...
==> Removing existing $pkgdir/ directory...
-> Profile data will be generated.
==> Starting build()...
==> Cleaning source code tree...
HEAD is now at a0ab2f162 build: Update to header 1.3.224
-- The C compiler identification is GNU 12.1.1
-- The CXX compiler identification is GNU 12.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
********************************************************************************
* NOTE: Not adding target to run update_deps.py automatically.                 *
********************************************************************************
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found VulkanHeaders: /usr/include  
-- Found VulkanRegistry: /usr/share/vulkan/registry  
-- Detected Vulkan Version 1.3.210
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Found Git: /usr/bin/git (found version "2.37.1") 
-- Found xcb: /usr/include  
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found WAYLAND_CLIENT: /usr/lib/libwayland-client.so  
-- Found WAYLAND_SERVER: /usr/lib/libwayland-server.so  
-- Found WAYLAND_EGL: /usr/lib/libwayland-egl.so  
-- Found WAYLAND_CURSOR: /usr/lib/libwayland-cursor.so  
-- Found WAYLAND: /usr/lib/libwayland-client.so;/usr/lib/libwayland-server.so;/usr/lib/libwayland-egl.so;/usr/lib/libwayland-cursor.so  
-- Looking for secure_getenv
-- Looking for secure_getenv - found
-- Looking for __secure_getenv
-- Looking for __secure_getenv - not found
-- The ASM compiler identification is GNU
-- Found assembler: /usr/lib/ccache/bin/cc
-- Looking for cet.h
-- Looking for cet.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/build64
[  5%] Building C object loader/CMakeFiles/asm_offset.dir/asm_offset.c.o
In file included from /home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/loader_common.h:40,
from /home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/asm_offset.c:26:
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:671:5: error: unknown type name 'PFN_vkCmdTraceRaysIndirect2KHR'
671 |     PFN_vkCmdTraceRaysIndirect2KHR CmdTraceRaysIndirect2KHR;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:881:5: error: unknown type name 'PFN_vkGetImageSubresourceLayout2EXT'
881 |     PFN_vkGetImageSubresourceLayout2EXT GetImageSubresourceLayout2EXT;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:930:5: error: unknown type name 'PFN_vkGetPipelinePropertiesEXT'
930 |     PFN_vkGetPipelinePropertiesEXT GetPipelinePropertiesEXT;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:954:5: error: unknown type name 'PFN_vkGetShaderModuleIdentifierEXT'
954 |     PFN_vkGetShaderModuleIdentifierEXT GetShaderModuleIdentifierEXT;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:955:5: error: unknown type name 'PFN_vkGetShaderModuleCreateInfoIdentifierEXT'
955 |     PFN_vkGetShaderModuleCreateInfoIdentifierEXT GetShaderModuleCreateInfoIdentifierEXT;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:958:5: error: unknown type name 'PFN_vkGetFramebufferTilePropertiesQCOM'
958 |     PFN_vkGetFramebufferTilePropertiesQCOM GetFramebufferTilePropertiesQCOM;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vv/PKGBUILDS/vulkan-icd-loader-git/src/Vulkan-Loader/loader/generated/vk_layer_dispatch_table.h:959:5: error: unknown type name 'PFN_vkGetDynamicRenderingTilePropertiesQCOM'
959 |     PFN_vkGetDynamicRenderingTilePropertiesQCOM GetDynamicRenderingTilePropertiesQCOM;
|     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [loader/CMakeFiles/asm_offset.dir/build.make:76: loader/CMakeFiles/asm_offset.dir/asm_offset.c.o]Error 1
make[1]: *** [CMakeFiles/Makefile2:155: loader/CMakeFiles/asm_offset.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by Tk-Glitch** [comment]
You'll need to first build and install https://github.com/Frogging-Family/vulkan-headers-git
Then rebuilding the icd-loader should go fine :)

**Comment by veganvelociraptor** [comment]
> You'll need to first build and install https://github.com/Frogging-Family/vulkan-headers-git Then rebuilding the icd-loader should go fine :)

Yep, already did that. I tried re-cloning the git repo and reinstalling both of them, the issue remains. :(

**Comment by veganvelociraptor** [comment]
Umm this is weird. I did it again. Last time I tried, it kept insisting that 1.3.210.r0.g0c59287-1 is the latest version and all makepkg -csi did was recompile and reinstall the very same version.

Now that I tried a git clone vulkan-headers-git in /tmp/ again, it downloaded and compiled 1.3.224.r1.gc896e2f-1.

..and now, vulkan-icd-loader-git compile as intended???

**Comment by Tk-Glitch** [comment]
I was about to say that if you're using the `PKGBUILDS` repo, you'll want to use the TkgThingy script to update the submodules, but since you're using the correct repo directly.. :frog:
The issue we had is that the active branch was changed from master to main, but the master branch was kept, frozen in time. So the codebase is old. I pushed the fix with https://github.com/Frogging-Family/vulkan-headers-git/commit/05282e33d13a53aa6cfd9eb275199e78c072b2ea 4 days ago. Maybe your clone was older than that and you didn't pull?
